src/HashBase.ts

```diff
@@ -123,25 +129,28 @@ export async function openHashBase(
 
       const prefix = hash.substring(0, prefixSize)
       const bucketPath = getBucketPath(databaseName, partition, prefix)
-      const setNewData = (oldData = {}): any =>
+      const setNewData = (oldData = {}): Promise<void> =>
         memlet.setJson(bucketPath, Object.assign(oldData, { [hash]: data }))
-      return memlet.getJson(bucketPath).then(
+      await memlet.getJson(bucketPath).then(
         bucketData => setNewData(bucketData),
-        // assuming bucket doesnt exist
+        // assuming bucket doesn't exist
         () => setNewData()
       )
     },
 
-    query(partition: string, hashes: string[]): Promise<any[]> {
+    query(partition: string, hashes: string[]) {
       return find(partition, hashes)
     },
 
-    delete(partition: string, hashes: string[]): Promise<any[]> {
+    delete(partition: string, hashes: string[]) {
       return find(partition, hashes, true)
     },
 
-    async dumpData(partition: string): Promise<any> {
-      const dump = (d: Disklet, data: any = {}): Promise<any> => {
+    async dumpData(
```

Comment:
Instead of getting the files at every level in the recursion, just gather the getText promises, and then await Promise.all() the promises. In other words, change DFS recursion to BFS (maybe?). Do this in a new PR.

Reply:
Later PR for all data types